src/content/github.js

```diff
@@ -1,122 +1,112 @@
-        if (window.location.href !== this._currentUrl) {
-          this._currentUrl = window.location.href;
-          this._initGitHubButton();
-
-          commentObserver.disconnect();
-          this._resetGithubCommentObserver(commentObserver);
-          this._resetGithubNewCommentObserver(newCommentObserver);
-        }
-      });
-      pageObserver.observe(document.querySelector("body"), {
-        childList: true
-      });
-      this._resetGithubCommentObserver(commentObserver);
-      this._resetGithubNewCommentObserver(newCommentObserver);
-      this._isGithubListenerAdded = true;
-    }
-
-    this._initGitHubButton();
+    this._observeURLChanges();
+    this._observeDOMChanges();
+    this._createButtons();
   }
 
-  _initGitHubButton() {
-    if (GITHUB_VALID_PATHNAMES.test(window.location.pathname)) {
-      this._createGithubPrettierButtons();
+  _observeURLChanges() {
+    if (this._urlIntervalId) {
+      window.clearInterval(this._urlIntervalId);
     }
-  }
 
-  _resetGithubNewCommentObserver(observer) {
-    const content = document.querySelector(".js-discussion");
+    this._urlIntervalId = window.setInterval(() => {
+      if (window.location.href === this._currentUrl) {
+        return;
+      }
+
+      this._currentUrl = window.location.href;
+      this._domObserver.disconnect();
 
-    if (content) {
-      observer.disconnect();
-      observer.observe(content, { childList: true, subtree: true });
-    }
+      if (!GITHUB_VALID_PATHNAMES.test(window.location.pathname)) {
+        return;
+      }
+
+      this._init();
+    }, URL_POLLING_INTERVAL);
   }
 
-  _resetGithubCommentObserver(observer) {
-    for (const elem of document.querySelectorAll(".timeline-comment-group")) {
-      observer.observe(elem, {
-        attributes: true,
-        childList: true,
-        subtree: true
-      });
+  _observeDOMChanges() {
+    if (!this._domObserver) {
+      // TODO: filter out mutations in the callback to improve performance.
+      this._domObserver = new MutationObserver(() => this._createButtons());
     }
 
     for (const elem of document.querySelectorAll(
-      ".js-diff-progressive-container"
+      OBSERVABLE_CONTAINERS.join(",")
     )) {
-      observer.observe(elem, {
+      this._domObserver.observe(elem, {
+        attributes: true,
         childList: true,
         subtree: true
       });
     }
   }
 
-  _createGithubPrettierButtons() {
+  _createButtons() {
     for (const button of this._seachForGithubButtons()) {
-      let prettierBtn = button.parentNode.querySelector(".prettier-btn");
-
-      if (!prettierBtn) {
-        prettierBtn = renderButton(button.parentNode, {
-          append: true,
-          classes: ["prettier-btn"],
-          refNode: button.innerText === SUBMIT_NEW_ISSUE ? button : null,
-          style: { "margin-right": "4px" }
-        });
+      const parentNode = button.parentNode;
+
+      if (
+        !isElementVisible(parentNode) ||
```

Comment:
This is what actually fixes the memory leak. Previously, this was rebinding the event handler repeatedly for existing Prettier buttons. This was most likely done because the button in the commit/PR diff views is actually rendered on the page in a non-visible container, and the binding doesn't work when it finally shows up. This differs the creation of the Prettier button until the node it will be rendered into is visible to user (it checks if the element is hidden and if it's in the viewport), and everything now gets bound once!